File: src/main/java/coffeetime/infrastructure/DefaultNicknameGenerator.java
package coffeetime.infrastructure;

import coffeetime.domain.DefaultNickname;
import java.util.List;
import java.util.Random;
import java.util.concurrent.ThreadLocalRandom;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class DefaultNicknameGenerator implements DefaultNickname {

	private final List<String> adjectives;
	private final List<String> nouns;

	@Override
	public String generate() {
		Random random = ThreadLocalRandom.current();
		String adjective = adjectives.get(random.nextInt(adjectives.size()));
		String noun = nouns.get(random.nextInt(nouns.size()));
		return adjective + " " + noun;
	}
}

File: src/main/kotlin/coffeetime/support/response/DefaultNicknameGenerator.kt
package coffeetime.support.response

import coffeetime.domain.DefaultNickname
import java.util.List
import java.util.Random
import java.util.concurrent.ThreadLocalRandom
import lombok.RequiredArgsConstructor

@RequiredArgsConstructor
class DefaultNicknameGenerator(
	private val adjectives: List<String>,
	private val nouns: List<String>
): DefaultNickname {

	override fun generate(): String {
		val random = ThreadLocalRandom.current()
		val adjective = adjectives[random.nextInt(adjectives.size)]
		val noun = nouns[random.nextInt(nouns.size)]
		return "$adjective $noun"
	}
}
